packages/block-library/src/columns/edit.js

```diff
@@ -264,4 +175,59 @@ export default withDispatch( ( dispatch, ownProps, registry ) => ( {
 
 		replaceInnerBlocks( clientId, innerBlocks, false );
 	},
-} ) )( ColumnsEdit );
+} ) )( ColumnsEditContainer );
+
+const createBlocksFromInnerBlocksTemplate = ( innerBlocksTemplate ) => {
+	return map(
+		innerBlocksTemplate,
+		( [ name, attributes, innerBlocks = [] ] ) =>
+			createBlock( name, attributes, createBlocksFromInnerBlocksTemplate( innerBlocks ) )
+	);
+};
+
+const ColumnsEdit = ( props ) => {
```

Comment:
So If I understand properly here, this component is not specific to the columns block at all and is meant to be moved to the framework right? Potentially, it can be reused in other blocks before that as well?

Reply:
Yes, that's the goal. I just don't think it's ready as-is.

I tried to make it as flexible as possible but it won't work in all cases as noted in my comment:

> At the moment the wrapper which handles the pattern selection is presented when there are no nested blocks added to the block. In the case of Columns block, it's perfectly fine as it always needs to have at least 2 nested blocks. However, this might not fit all use cases as some blocks might assume that there might be no blocks provided when the block gets created. In addition, it doesn't cover also blocks that don't have inner blocks where we want to use Patterns API to offer some initial setup.

I bet this can be further abstracted by introducing a prop that performs the check whether the patterns picker should be rendered.